### PR TITLE
Fix testing of the galaxy-tool-shed package

### PR DESCRIPTION
**Pull request by nsoranzo**
It was not tested before because of the missing newline at the end of `packages/packages_by_dep_dag.txt` and of how `read` works in bash.

Also:
- Allow empty lines in packages_by_dep_dag.txt and divide it into layers, so that packages in each layer depend only on packages in the layers above.
- Fix type annotations and use of re-exported imports.
- Fix various requirements of packages.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: packages/packages_by_dep_dag.txt [hunk 1]
```diff
@@ -1,21 +1,27 @@
 util
-schema
+
 config
 files
 job_metrics
 navigation
 objectstore
+schema
```
Comment (jmchilton) [inline packages/packages_by_dep_dag.txt:8]: Conceptually I preferred this package higher to be more explicit about it not having a lot of dependencies. But you've broken it up here into like levels and alphabetized them... I guess this is okay. The spaces are certainly helpful and appreciated. Thank you

File: packages/web_stack/setup.cfg [hunk 2]
```diff
@@ -32,6 +32,7 @@ version = 23.1.dev0
 [options]
 include_package_data = True
 install_requires =
+    galaxy-data
```
Comment (jmchilton) [inline packages/web_stack/setup.cfg:35]: I don't like this - but handlers.py has a ``galaxy.model.base import transaction`` import so it does accurately reflect what is going on. Thank you for the fix.

**Comment by jmchilton** [comment]
This is really fantastic - thank you for your constant diligence.

**Review by jmchilton — APPROVED** [review]

File: packages/packages_by_dep_dag.txt (hunk 1 above)
Comment (nsoranzo) [reply packages/packages_by_dep_dag.txt:8]: Yes, for me at least, it makes it easier to visualise that the packages in this group form a layer that only depend on the above layers, in this case `util`.

File: packages/web_stack/setup.cfg (hunk 2 above)
Comment (nsoranzo) [reply packages/web_stack/setup.cfg:35]: And `from galaxy.model import database_utils` in `lib/galaxy/web_stack/__init__.py`

**Pull request merged**